### PR TITLE
Unordered staging

### PR DESCRIPTION
**Pull request by scuy**
The object of the task chain will no longer be defined by the name of
the folder in the staging area. Instead, the files that should be
handled in a job will be given to retrieve_from_staging as a parameter.

Therefore there is no longer a need for providing an object_id when
creating a job. Depending on the job type the id can now be generated
or read from the metadata as part of the job chain.

Since retrieve_staging needs an id under which the result of the chain
is published it uses the job_id until proper mechanisms for reading
the object id are in place.

In addition to being able to provide the names of the files that will
be retrieved from the staging area, provided folders will also be
copied (recursively) to the working directory. Files from folders given in the paths parameter are copied to the root of the job directory in the workspace.

Also, the JSON metadata file is now expected to be at the root of the job
directory in the workspace.

### CODE REVIEW

File: workers/default/repository/tasks.py [hunk 1]
```diff
@@ -9,16 +9,37 @@
...
+
 class RetrieveFromStagingTask(BaseTask):
 
     name = "retrieve_from_staging"
 
     def execute_task(self):
-        staging_path = os.path.join(staging_dir, self.object_id)
+        staging_path = os.path.join(staging_dir)
+
+        # TODO: move to base class
```
Comment (ngrippa) [inline workers/default/repository/tasks.py]: Wieso hast du das TODO hier drin gelassen?

Comment (scuy) [reply workers/default/repository/tasks.py]: Vergessen ... :)

**Comment by ngrippa** [comment]
Meinen Segen hat die PR jetzt. Will sich das noch jemand anschauen und dann mergen? @rhenck ?
Und: hatten wir nicht gesagt wir lassen das mit den Forks sondern machen das direkt in feature branches? Dann ist das lokale auschecken ein bisschen einfacher.

**Comment by scuy** [comment]
Stimmt, mein upstream war noch auf scuy eingestellt, hatte ich garnicht gemerkt.

**Pull request merged**